www/apps/resources/app/commerce-modules/auth/auth-flows/page.mdx

```diff
@@ -77,46 +91,63 @@ Check out the [AuthIdentity](/references/auth/models/AuthIdentity) reference for
+    // ...
+  }
+)
+
+if (location) {
+  // return the location for the front-end to redirect to
+}
+
+if (!success) {
+  // authentication failed
 }
+
+// authentication successful
 ```
 
-It means the authentication process requires the user to perform an action with a third-party service. For example, when using the `google` provider, the user goes to the URL specified in the `location`'s value to log in with their Google account. 
+If the `authenticate` method returns a `location` property, the authentication process requires the user to perform an action with a third-party service. So, you return the `location` to the front-end or client to redirect to that URL.
+
+For example, when using the `google` provider, the `location` is the URL that the user is navigated to login.
+
+If the `authenticate` method doesn't return a `location` property and the returned `success` property, the user is successfully authenticated. This happens after the user logs-in with the provider, such as google, the first time.
 
 ![Diagram showcasing the first part of the third-party authentication flow](https://res.cloudinary.com/dza7lstvk/image/upload/v1711374847/Medusa%20Resources/third-party-auth-1_enyedy.jpg)
 
 ### validateCallback
 
 Providers handling this authentication flow must implement the `validateCallback` method. It implements the logic to validate the authentication with the third-party service.
 
-So, once the user performs the required action, the third-party service must redirect to an API route that uses the [validateCallback method of the Auth Module's main service](/references/auth/validateCallback). The method calls the specified provider’s `validateCallback` method passing it the authentication details it received in the second parameter:
+So, once the user performs the required action with the third-party service (for example, log-in with Google), the frontend must redirect to an API route that uses the [validateCallback method of the Auth Module's main service](/references/auth/validateCallback).
 
-```ts
-const data = await authModuleService.validateCallback(
-    "google",
-    // passed to auth provider
-    {
-      // ...
-    }
-  )
-```
-
-If the authentication is successful, the `validateCallback` method returns the same data as the basic authentication:
+The method calls the specified provider’s `validateCallback` method passing it the authentication details it received in the second parameter:
 
 ```ts
-data = {
-	success: true,
-	authIdentity: {
-    // ...
-  },
+const { success, authIdentity } = await authModuleService.validateCallback(
+  "google",
+  // passed to auth provider
+  {
+    // request data, such as
+    url,
+    headers,
+    query,
+    body,
+    protocol,
+  }
+)
+
+if (success) {
+  // authentication succeeded
 }
 ```
 
+If the returned `success` property is `true`, the authentication with the third-party provider was successful. Subsequent authentication attempts can use the `authenticate` method only.
```

Comment:
Again, callback has to happen every time an oauth is triggered, `authenticate` is only there to construct the initial redirect URL